l10n_fi_reconcile/tests/test_reconciliation_proposition.py

```diff
@@ -0,0 +1,141 @@
+    {'name': 'partial_match1', 'payment_reference': REFERENCE_TWO_WITH_ZEROS, 'amount': AMOUNT2},
+    {'name': 'partial_match2', 'payment_reference': REFERENCE_TWO_WITH_ZEROS, 'amount': AMOUNT3},
+]
+# account.bank.statement.line
+# We have to create the account.move.lines in a way that doesn't allow us to (easily) know their ids, so we
+# use the move line's name to compare the return values.
+BANK_STATEMENT_LINES = [
+    {'reference': REFERENCE_ONE_WITH_ZEROS,     'amount': AMOUNT1, 'expected_match': 'perfect_match1'},
+    {'reference': REFERENCE_ONE_WITHOUT_ZEROS,  'amount': AMOUNT2, 'expected_match': 'perfect_match2'},
+    {'reference': REFERENCE_TWO_WITH_ZEROS,     'amount': AMOUNT1, 'expected_match': 'partial_match1'},
+]
+
+
+class ReconciliationPropositionTestCase(TransactionCase):
+    def setUp(self):
+        super(ReconciliationPropositionTestCase, self).setUp()
+
+        self.company = self.env['res.company'].search([], limit=1)
+        self.assertTrue(self.company)
+        # the method also matches by user's company, so we must make sure the correct one is linked to it
+        self.env.user.company_id = self.company.id
+
+        self.journal = self.env['account.journal'].search([('type', '=', 'sale')], limit=1)
+        self.assertTrue(self.journal)
+
+        # any account that can be reconciled
+        self.account = self.env['account.account'].search([('reconcile', '=', True)], limit=1)
+        self.assertTrue(self.account)
+
+        self.partner = self.env['res.partner'].search([('customer', '=', True)], limit=1)
+        self.assertTrue(self.partner)
+        # invoices are there just to hold payment_reference which move lines are related to
+        # place the invoices in a dict with payment reference as the key
+        invoices = dict()
+        invoices[REFERENCE_ONE_WITH_ZEROS] = self.env['account.invoice'].create({
+            'partner_id': self.partner.id,
+            'account_id': self.account.id,
+            'journal_id': self.journal.id,
+            'company_id': self.company.id,
+            'payment_reference': REFERENCE_ONE_WITH_ZEROS,
+        })
+        invoices[REFERENCE_TWO_WITH_ZEROS] = self.env['account.invoice'].create({
+            'partner_id': self.partner.id,
+            'account_id': self.account.id,
+            'journal_id': self.journal.id,
+            'company_id': self.company.id,
+            'payment_reference': REFERENCE_TWO_WITH_ZEROS,
+        })
+
+        self.account_bank_statement_id = self.env['account.bank.statement'].create({
+            'journal_id': self.journal.id,
+            'company_id': self.company.id,
+        })
+
+        # We need a payment_method to assign to payments. This is how account.invoice
+        # gets payment_method when creating payments
+        payment_method = self.env.ref('account.account_payment_method_manual_in')
+        self.assertTrue(payment_method)
```

Comment:
`ref` raises a ValueError if an id isn't found, so we don't need this separate check (even though I totally told you to put these in, sorry).